From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: FUKAUMI Naoki <naoki@radxa.com>
Date: Sun, 9 Nov 2025 12:55:27 +0000
Subject: FROMLIST(v7): arm64: dts: rockchip: Trivial changes for Radxa ROCK 5C

No functional change.

Signed-off-by: FUKAUMI Naoki <naoki@radxa.com>
---
Changes in v7:
- Reflect other commits, no functional change
Changes in v6:
- New
---
 arch/arm64/boot/dts/rockchip/rk3588s-rock-5c.dts | 64 +++++-----
 1 file changed, 31 insertions(+), 33 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3588s-rock-5c.dts b/arch/arm64/boot/dts/rockchip/rk3588s-rock-5c.dts
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588s-rock-5c.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588s-rock-5c.dts
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: (GPL-2.0+ OR MIT)
 /*
- * Copyright (c) 2024 Radxa Computer (Shenzhen) Co., Ltd.
+ * Copyright (c) 2024,2025 Radxa Computer (Shenzhen) Co., Ltd.
  */
 
 /dts-v1/;
@@ -25,7 +25,7 @@ chosen {
 		stdout-path = "serial2:1500000n8";
 	};
 
-	analog-sound {
+	sound {
 		compatible = "audio-graph-card";
 		label = "rk3588-es8316";
 		dais = <&i2s0_8ch_p0>;
@@ -49,14 +49,14 @@ hdmi0_con_in: endpoint {
 
 	leds {
 		compatible = "gpio-leds";
-		pinctrl-names = "default";
-		pinctrl-0 = <&led_pins>;
 
 		led-0 {
 			color = <LED_COLOR_ID_GREEN>;
 			default-state = "on";
 			function = LED_FUNCTION_POWER;
 			gpios = <&gpio3 RK_PC4 GPIO_ACTIVE_HIGH>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&power_led>;
 		};
 
 		led-1 {
@@ -65,6 +65,8 @@ led-1 {
 			function = LED_FUNCTION_STATUS;
 			gpios = <&gpio3 RK_PD5 GPIO_ACTIVE_HIGH>;
 			linux,default-trigger = "heartbeat";
+			pinctrl-names = "default";
+			pinctrl-0 = <&io_led>;
 		};
 	};
 
@@ -76,19 +78,19 @@ fan: fan {
 		pwms = <&pwm3 0 60000 0>;
 	};
 
-	pcie2x1l2_3v3: regulator-pcie2x1l2-3v3 {
+	vcc3v3_pcie2x1l2: regulator-3v3-0 {
 		compatible = "regulator-fixed";
 		enable-active-high;
 		gpio = <&gpio0 RK_PC5 GPIO_ACTIVE_HIGH>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&pow_en>;
-		regulator-name = "pcie2x1l2_3v3";
+		regulator-name = "vcc3v3_pcie2x1l2";
 		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
 		vin-supply = <&vcc_sysin>;
 	};
 
-	vcc5v_dcin: regulator-vcc5v-dcin {
+	vcc5v_dcin: regulator-5v0-0 {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc5v_dcin";
 		regulator-always-on;
@@ -97,7 +99,7 @@ vcc5v_dcin: regulator-vcc5v-dcin {
 		regulator-max-microvolt = <5000000>;
 	};
 
-	vcc5v0_usb_host: regulator-vcc5v0-usb-host {
+	vcc5v0_usb_host: regulator-5v0-1 {
 		compatible = "regulator-fixed";
 		enable-active-high;
 		gpio = <&gpio4 RK_PB5 GPIO_ACTIVE_HIGH>;
@@ -109,7 +111,7 @@ vcc5v0_usb_host: regulator-vcc5v0-usb-host {
 		vin-supply = <&vcc_sysin>;
 	};
 
-	vcc5v0_usb_otg0: regulator-vcc5v0-usb-otg0 {
+	vcc5v0_usb_otg0: regulator-5v0-2 {
 		compatible = "regulator-fixed";
 		enable-active-high;
 		gpio = <&gpio0 RK_PD4 GPIO_ACTIVE_HIGH>;
@@ -121,7 +123,7 @@ vcc5v0_usb_otg0: regulator-vcc5v0-usb-otg0 {
 		vin-supply = <&vcc_sysin>;
 	};
 
-	vcc_1v1_nldo_s3: regulator-vcc-1v1-nldo-s3 {
+	vcc_1v1_nldo_s3: regulator-1v1 {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc_1v1_nldo_s3";
 		regulator-always-on;
@@ -131,17 +133,7 @@ vcc_1v1_nldo_s3: regulator-vcc-1v1-nldo-s3 {
 		vin-supply = <&vcc_sysin>;
 	};
 
-	vcc_3v3_pmu: regulator-vcc-3v3-pmu {
-		compatible = "regulator-fixed";
-		regulator-name = "vcc_3v3_pmu";
-		regulator-always-on;
-		regulator-boot-on;
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		vin-supply = <&vcc_3v3_s3>;
-	};
-
-	vcc_3v3_s0: regulator-vcc-3v3-s0 {
+	vcc_3v3_s0: regulator-3v3-1 {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc_3v3_s0";
 		regulator-always-on;
@@ -151,7 +143,7 @@ vcc_3v3_s0: regulator-vcc-3v3-s0 {
 		vin-supply = <&vcc_3v3_s3>;
 	};
 
-	vcc_5v0: regulator-vcc-5v0 {
+	vcc_5v0: regulator-5v0-3 {
 		compatible = "regulator-fixed";
 		enable-active-high;
 		gpio = <&gpio4 RK_PA3 GPIO_ACTIVE_HIGH>;
@@ -163,7 +155,7 @@ vcc_5v0: regulator-vcc-5v0 {
 		vin-supply = <&vcc_sysin>;
 	};
 
-	vcc_sysin: regulator-vcc-sysin {
+	vcc_sysin: regulator-5v0-4 {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc_sysin";
 		regulator-always-on;
@@ -173,7 +165,7 @@ vcc_sysin: regulator-vcc-sysin {
 		vin-supply = <&vcc5v_dcin>;
 	};
 
-	vcca: regulator-vcca {
+	vcca: regulator-4v0 {
 		compatible = "regulator-fixed";
 		regulator-name = "vcca";
 		regulator-always-on;
@@ -183,7 +175,7 @@ vcca: regulator-vcca {
 		vin-supply = <&vcc_sysin>;
 	};
 
-	vdd_3v3: regulator-vdd-3v3 {
+	vdd_3v3: regulator-3v3-2 {
 		compatible = "regulator-fixed";
 		enable-active-high;
 		gpio = <&gpio0 RK_PA0 GPIO_ACTIVE_HIGH>;
@@ -203,6 +195,7 @@ &combphy0_ps {
 };
 
 &combphy2_psu {
+	phy-supply = <&vcc5v0_usb_host>;
 	status = "okay";
 };
 
@@ -435,7 +428,7 @@ &pcie2x1l2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pcie20x1_2_perstn_m0>;
 	reset-gpios = <&gpio3 RK_PD1 GPIO_ACTIVE_HIGH>;
-	vpcie3v3-supply = <&pcie2x1l2_3v3>;
+	vpcie3v3-supply = <&vcc3v3_pcie2x1l2>;
 	status = "okay";
 };
 
@@ -445,13 +438,16 @@ &pd_gpu {
 
 &pinctrl {
 	leds {
-		led_pins: led-pins {
-			rockchip,pins = <3 RK_PC4 RK_FUNC_GPIO &pcfg_pull_none>,
-					<3 RK_PD5 RK_FUNC_GPIO &pcfg_pull_none>;
+		io_led: io-led {
+			rockchip,pins = <3 RK_PD5 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+
+		power_led: power-led {
+			rockchip,pins = <3 RK_PC4 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
 
-	mdio {
+	ethernet {
 		gmac1_rstn: gmac1-rstn {
 			rockchip,pins = <3 RK_PB7 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
@@ -485,7 +481,9 @@ usb_otg_pwren_h: usb-otg-pwren-h {
 		usb_wifi_pwr: usb-wifi-pwr {
 			rockchip,pins = <0 RK_PA0 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
+	};
 
+	regulators {
 		vcc_5v0_pwren_h: vcc-5v0-pwren-h {
 			rockchip,pins = <4 RK_PA3 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
@@ -512,7 +510,7 @@ &sdhci {
 	no-sd;
 	non-removable;
 	vmmc-supply = <&vcc_3v3_s0>;
-	vqmmc-supply = <&vcc_1v8_s3>;
+	vqmmc-supply = <&vccio_flash>;
 	status = "okay";
 };
 
@@ -688,7 +686,7 @@ regulator-state-mem {
 				};
 			};
 
-			vcc_3v3_s3: dcdc-reg8 {
+			vcc_3v3_pmu: vcc_3v3_s3: dcdc-reg8 {
 				regulator-name = "vcc_3v3_s3";
 				regulator-always-on;
 				regulator-boot-on;
@@ -711,7 +709,7 @@ regulator-state-mem {
 				};
 			};
 
-			vcc_1v8_s3: dcdc-reg10 {
+			vccio_flash: vcc_1v8_s3: dcdc-reg10 {
 				regulator-name = "vcc_1v8_s3";
 				regulator-always-on;
 				regulator-boot-on;
-- 
Armbian

